### PR TITLE
feat(rds): specify PreferredMaintenanceWindow in reader or writer props

### PR DESCRIPTION
**Pull request by Konosh93**
### Issue #29687 

Closes ##29687 

### Reason for this change

Since `instanceProps` is deprecated it is better to allow setting `PreferredMaintenanceWindow` using writer and readers props.

### Description of changes

Added optional preferredMaintenanceWindow field to ClusterInstanceOptions and added a unit test.

### Description of how you validated changes

- Add a single unit test

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by Konosh93** [comment]
[IMO] Based on the previous fix, I think this PR also qualifies for an Exemption Request.

https://github.com/aws/aws-cdk/pull/29033#issuecomment-1934602020

File: packages/aws-cdk-lib/aws-rds/README.md [hunk 1]
```diff
@@ -1224,3 +1224,15 @@ new rds.DatabaseCluster(this, 'DatabaseCluster', {
   preferredMaintenanceWindow: 'Sat:22:15-Sat:22:45',
 });
 ```
+You can also use writer or readers props like:
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-rds/README.md:1227]: nit: "You can also set the preferred maintenance window via reader and writer props:"

File: packages/aws-cdk-lib/aws-rds/test/cluster.test.ts [hunk 2]
```diff
@@ -226,6 +226,29 @@ describe('cluster new api', () => {
...
+
+      const PREFERRED_MAINTENANCE_WINDOW: string = 'Sun:12:00-Sun:13:00';
+
+      // WHEN
+      new DatabaseCluster(stack, 'Database', {
+        engine: DatabaseClusterEngine.AURORA,
+        vpc: vpc,
+        writer: ClusterInstance.provisioned('Instance1', {
+          preferredMaintenanceWindow: PREFERRED_MAINTENANCE_WINDOW,
+        }),
```
Comment (msambol) [inline packages/aws-cdk-lib/aws-rds/test/cluster.test.ts:249]: can you add a test for `reader` too?

**Comment by msambol** [comment]
Can you change the title to:

"feat(rds): specify PreferredMaintenanceWindow in reader or writer props" 

Also, this one should have an integration test. Let me know if you need help with that.

File: packages/aws-cdk-lib/aws-rds/README.md (hunk 1 above)
Comment (Konosh93) [reply packages/aws-cdk-lib/aws-rds/README.md:1227]: Sounds good thanks

File: packages/aws-cdk-lib/aws-rds/test/cluster.test.ts (hunk 2 above)
Comment (Konosh93) [reply packages/aws-cdk-lib/aws-rds/test/cluster.test.ts:249]: Sure.

**Comment by Konosh93** [comment]
> Can you change the title to:
> 
> "feat(rds): specify PreferredMaintenanceWindow in reader or writer props"

Sure. Thanks. 👍 

> Also, this one should have an integration test. Let me know if you need help with that.

I modified an existing test in `packages/@aws-cdk-testing/framework-integ/test/aws-rds/test/integ.cluster.ts`.

**Review by msambol — APPROVED** [review]
LGTM

**Review by godwingrs22 — APPROVED** [review]
LGTM. Thanks @Konosh93 for your contribution and @msambol for your review.

**Pull request merged**